src/solver/simulation/hydro-remix-new.cpp

```diff
@@ -0,0 +1,299 @@
+                   const std::vector<double>& new_D,
+                   const std::vector<double>& new_H,
+                   const std::vector<int>& tried_creux,
+                   const std::vector<double>& P_max,
+                   const std::vector<bool>& filter_hours_remix,
+                   double top)
+{
+    double min_val = top;
+    int min_idx = -1;
+    for (int i = 0; i < G_plus_H.size(); ++i)
+    {
+        if (new_D[i] > 0 && new_H[i] < P_max[i] && tried_creux[i] == 0 && filter_hours_remix[i])
+        {
+            if (G_plus_H[i] < min_val)
+            {
+                min_val = G_plus_H[i];
+                min_idx = i;
+            }
+        }
+    }
+    return min_idx;
+}
+
+int find_max_index(const std::vector<double>& G_plus_H,
+                   const std::vector<double>& new_H,
+                   const std::vector<int>& tried_pic,
+                   const std::vector<double>& P_min,
+                   const std::vector<bool>& filter_hours_remix,
+                   double ref_value,
+                   double eps)
+{
+    double max_val = 0;
+    int max_idx = -1;
+    for (int i = 0; i < G_plus_H.size(); ++i)
+    {
+        if (new_H[i] > P_min[i] && G_plus_H[i] >= ref_value + eps && tried_pic[i] == 0
+            && filter_hours_remix[i])
+        {
+            if (G_plus_H[i] > max_val)
+            {
+                max_val = G_plus_H[i];
+                max_idx = i;
+            }
+        }
+    }
+    return max_idx;
+}
+
+static bool isLessThan(const std::vector<double>& a, const std::vector<double>& b)
+{
+    std::vector<double> a_minus_b;
+    std::ranges::transform(a, b, std::back_inserter(a_minus_b), std::minus<double>());
+    return std::ranges::all_of(a_minus_b, [](const double& e) { return e <= 0.; });
+}
+
+static bool isLessThan(const std::vector<double>& v, const double c)
+{
+    return std::ranges::all_of(v, [&c](const double& e) { return e <= c; });
+}
+
+static bool isGreaterThan(const std::vector<double>& v, const double c)
+{
+    return std::ranges::all_of(v, [&c](const double& e) { return e >= c; });
+}
+
+static void checkInputCorrectness(const std::vector<double>& G,
+                                  const std::vector<double>& H,
+                                  const std::vector<double>& D,
+                                  const std::vector<double>& levels,
+                                  const std::vector<double>& P_max,
+                                  const std::vector<double>& P_min,
+                                  double initial_level,
+                                  double capacity,
+                                  const std::vector<double>& inflows,
+                                  const std::vector<double>& overflow,
+                                  const std::vector<double>& pump,
+                                  const std::vector<double>& S,
+                                  const std::vector<double>& DTG_MRG)
```

Comment:
1. I don't really agree with having this function `static`, it might be public to ease testing
2. Please avoid single character variable names

Reply:
1. If this function was a method of a class, it would be **private**. Question is : do we expose it to the outside world ? Well literature says : test classes / objects behaviors, not their methods, and especially their **private** methods.  We can't make every **private** methods **public** just to test them. So here, we test how the algorithm reacts to incorrect input data, and doing this
2. I do agree